src/plugins/dataflow-tool/components/ui/dataflow-program-topbar.tsx

```diff
@@ -1,137 +1,131 @@
+//         <div className="progress-time">
+//         {`${props.hours}:${props.minutes}:${props.seconds}`}
+//         </div>
+//       </div>
+//     </div>
+//   );
+// };
 
-interface DurationSelectorProps {
-  onRunProgramClick: () => void;
-  programRunTimes: ProgramRunTime[];
-  programDefaultRunTime: number;
-  onProgramTimeSelectClick: (type: number) => void;
-  isRunEnabled: boolean;
+interface RateSelectorProps {
+  rateOptions: ProgramDataRate[];
+  dataRate: number;
+  onRateSelectClick: (rate: number) => void;
   readOnly: boolean;
 }
 
-const DurationSelectorComponent: React.SFC<DurationSelectorProps> = (props: DurationSelectorProps) => {
+const RateSelectorComponent: React.FC<RateSelectorProps> = (props: RateSelectorProps) => {
   const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    props.onProgramTimeSelectClick(Number(event.target.value));
+    props.onRateSelectClick(Number(event.target.value));
   };
   return (
     <div className="running-container">
-      <div className="duration" title="Set Program Duration">
+      <div className="datarate" title="Set Program Data Rate">
         <div className="label-back">
-          <div className="label">Duration</div>
+          <label className="label" htmlFor="rate-select">Data Rate</label>
         </div>
-        <div className="duration-options-back">
-          <div className="duration-options">
+        <div className="datarate-options-back">
+          <div className="datarate-options">
             <select onChange={handleSelectChange}
-              disabled={!props.isRunEnabled || props.readOnly}
-              value={props.programDefaultRunTime.toString()}
+              disabled={props.readOnly}
+              value={props.dataRate.toString()}
+              id="rate-select"
             >
-              { props.programRunTimes.map((rt: ProgramRunTime) => (
-                  <option key={rt.text} value={rt.val} disabled={rt.disabled}>
-                    {rt.text}
-                  </option>
-                ))
-              }
+              { props.rateOptions.map((rate: ProgramDataRate) => (
+                <option key={rate.text} value={rate.val} disabled={rate.disabled}>
+                  {rate.text}
+                </option>
+              ))}
             </select>
           </div>
         </div>
       </div>
-      <button
-        className="program-state-button"
-        title="Run Program"
-        onClick={props.onRunProgramClick}
-        disabled={!props.isRunEnabled || props.readOnly}
-      >
-        <div className="icon run" />
-        <div className="text">Run</div>
-      </button>
     </div>
   );
 };
 
+interface RecordButtonProps {
+  readOnly: boolean;
+}
+
+const RecordButton: React.FC<RecordButtonProps> = (props: RecordButtonProps) => {
+  return (
+    <button
+      className="program-state-button"
+      title="Record"
+      onClick={() => { return null; }}
```

Comment:
```suggestion
      onClick={() => null}
```
Simpler way to achieve the same effect